config.go

```diff
@@ -301,13 +301,18 @@ func getSettings(ctx context.Context, includeOverrides, includeParameterStore bo
 		}
 
 		readAdminSecrets(ctx, paramMgr, settingsValue, settingsType, "", paramCache, adminCatcher)
-		if adminCatcher.HasErrors() {
+		if adminCatcher.HasErrors() && ctx.Err() == nil {
 			grip.Error(errors.Wrap(adminCatcher.Resolve(), "reading admin settings in parameter store"))
 		} else {
 			baseConfig = paramConfig
 		}
 	}
 
+	// The context may be cancelled while getting settings.
+	// In this case, there's no need to return an error.
+	if ctx.Err() != nil {
```

Comment:
this will potentially hide other non-context errors. I don't know if that's a problem though if the context is also canceled

Reply:
I think it's okay because the context cancel should take precedence